db/sql/sql.go

```diff
@@ -1115,6 +1316,17 @@ func (tx *Transaction) LockFetchContext(ctx context.Context, s *schema.Schema, f
 	return list[0], nil
 }
 
+func (tx *CachedTransaction) LockFetchContext(ctx context.Context, s *schema.Schema, filter transaction.Filter, lockPolicy schema.LockPolicy, options *transaction.ViewOptions) (*schema.Resource, error) {
+	list, _, err := tx.LockListContext(ctx, s, filter, nil, nil, lockPolicy)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to fetch and lock %s: %s", filter, err)
+	}
+	if len(list) < 1 {
+		return nil, transaction.ErrResourceNotFound
+	}
+	return list[0], nil
+}
+
```

Comment:
this file is more than 1500 lines now. could we move `CachedTransaction` to a separate file?